PureBasicIDE/ColorSchemes.pb

```diff
@@ -0,0 +1,324 @@
+      EndIf
+    CompilerEndIf
+    
+    For i = 0 To #COLOR_Last
+      If (Colors(i)\PrefsValue >= 0)
+        UpdatePreferenceSyntaxColor(i, Colors(i)\PrefsValue)
+      Else
+        Colors(i)\PrefsValue = GuessColorSchemeColor(*ColorScheme, i)
+        UpdatePreferenceSyntaxColor(i, Colors(i)\PrefsValue)
+      EndIf
+    Next i
+    
+    DisableSelectionColorGadgets(*ColorScheme)
+    
+    If (PreferenceToolsPanelFrontColor < 0)
+      PreferenceToolsPanelFrontColor = GuessColorSchemeColor(*ColorScheme, #COLOR_ToolsPanelFrontColor)
+    EndIf
+    If (PreferenceToolsPanelBackColor < 0)
+      PreferenceToolsPanelBackColor = GuessColorSchemeColor(*ColorScheme, #COLOR_ToolsPanelBackColor)
+    EndIf
+    
+    If IsImage(#IMAGE_Preferences_ToolsPanelFrontColor)
+      UpdateImageColorGadget(#GADGET_Preferences_ToolsPanelFrontColor, #IMAGE_Preferences_ToolsPanelFrontColor, PreferenceToolsPanelFrontColor)
+    EndIf
+    If IsImage(#IMAGE_Preferences_ToolsPanelBackColor)
+      UpdateImageColorGadget(#GADGET_Preferences_ToolsPanelBackColor, #IMAGE_Preferences_ToolsPanelBackColor, PreferenceToolsPanelBackColor)
+    EndIf
+    
+  EndIf
+EndProcedure
+
+; Find and remove a known color scheme by its name
+Procedure RemoveColorSchemeIfExists(Name.s)
+  If (Name <> "")
+    ForEach ColorScheme()
+      If (ColorScheme()\Name = Name)
+        DeleteElement(ColorScheme())
+        Break
+      EndIf
+    Next
+  EndIf
+EndProcedure
+
+; Read the specified *ColorScheme from data section (for built-in schemes)
+Procedure.i ReadColorSchemeFromDataSection(*ColorScheme.ColorSchemeStruct)
+  If (*ColorScheme)
+    ; This assumes the NAME STRING data has already been read!
+    With *ColorScheme
+      \File = ""
+      Read.l \ColorValue[#COLOR_ToolsPanelFrontColor]
+      Read.l \ColorValue[#COLOR_ToolsPanelBackColor]
+      For i = 0 To #COLOR_Last
+        Read.l \ColorValue[i]
+      Next i
+    EndWith
+  EndIf
+  
+  ProcedureReturn (*ColorScheme)
+EndProcedure
+
+; Load the specified *ColorScheme from file on disk (for external schemes)
+Procedure.i LoadColorSchemeFromFile(*ColorScheme.ColorSchemeStruct, File.s)
+  Protected Result.i = #Null
+  
+  If (File)
+    ; Basic validation of color scheme file...
+    If (OpenPreferences(File))
+      Name.s = GetFilePart(File, #PB_FileSystem_NoExtension)
+      If (#True);(PreferenceGroup("Sections") And (ReadPreferenceLong("IncludeColors", 0) = 1))
+        If (PreferenceGroup("Colors"))
+          
+          If (*ColorScheme) ; struct already specified - part of InitColorSchemes() list
+            ; Intentionally overwrite schemes of existing names - allows you to tweak the default themes, if desired
+            RemoveColorSchemeIfExists(Name)
+          Else ; NULL --> dynamically allocate a struct now - NOT part of InitColorSchemes() list!
+            *ColorScheme = AllocateStructure(ColorSchemeStruct)
+          EndIf
+          
+          If (*ColorScheme)
+            With *ColorScheme
```

Comment:
For better readability, the `With` statement should be avoided, especially in larger code blocks.

Reply:
OK